### ISSUE TITLE
[Android] [ToolbarAndroid] Empty error screen when using {uri: string} as logo or navIcon source

### ISSUE THREAD

**Issue by kusamakura** [facebook/react-native#3140]
React Native throws an undefined error when using `{uri: string}` for ToolbarAndroid's logo or navIcon props.

Sample runnable code:

```javascript
'use strict';

var React = require('react-native');
var {AppRegistry, ToolbarAndroid,} = React;

var ReactNativeUIExplorer = React.createClass({
  render: function() {
    return <ToolbarAndroid
      navIcon={{uri: 'http://facebook.github.io/react/img/logo_og.png'}}
      style={{height: 56}}
      title="Toolbar"
    />;
  }
});

AppRegistry.registerComponent('ReactNativeUIExplorer', () => ReactNativeUIExplorer);
```

![Emulator screen](https://cloud.githubusercontent.com/assets/882152/10189216/813c4c92-6797-11e5-970d-4c83e2a70779.png)
![Debugger console](https://cloud.githubusercontent.com/assets/882152/10189218/849c5cce-6797-11e5-8590-10664e882983.png)

**Comment by astreet** [comment]
@foghina is this a supported format of this prop?
@frantic any idea why we're throwing undefined?

**Comment by astreet** [comment]
(Thanks for the report!)

**Comment by foghina** [comment]
@kusamakura the toolbar only supports [static resources][0]. We are adding remote image support but it should only be used in dev mode to request images from the packager. Released apps should always have these assets bundled into the APK.

@frantic `ToolbarAndroid` should be [throwing an exception][1], not sure why it gets borked like that. I'm guessing it's because it's throwing a string and not an `Error`?

[0]: https://facebook.github.io/react-native/docs/image.html#adding-static-resources-to-your-android-app
[1]: https://github.com/facebook/react-native/blob/master/Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js#L119

**Comment by kusamakura** [comment]
@foghina Yeah, figured that was the case, but the error was weird.

Also, oot: I think using Image.propTypes.source for logo and navIcon makes them required, hence the warnings in my second screenshot!

**Comment by frantic** [comment]
Should be resolved by 5ca5ec76889040c7c65b4a73bfe42f6e675eb0bc, I'll let @foghina close

**Comment by foghina** [comment]
@kusamakura I have noticed some props have become required although `isRequired` is not specified. I'll look into it.

@frantic yep.
